crates/polars-plan/src/dsl/function_expr/temporal.rs

```diff
@@ -95,91 +95,60 @@ pub(super) fn datetime(
     if day.len() < max_len {
         day = day.new_from_index(0, max_len);
     }
-    let day = day.u32()?;
+    let day = day.i8()?;
 
-    let mut hour = hour.cast(&DataType::UInt32)?;
+    let mut hour = hour.cast(&DataType::Int8)?;
     if hour.len() < max_len {
         hour = hour.new_from_index(0, max_len);
     }
-    let hour = hour.u32()?;
+    let hour = hour.i8()?;
 
-    let mut minute = minute.cast(&DataType::UInt32)?;
+    let mut minute = minute.cast(&DataType::Int8)?;
     if minute.len() < max_len {
         minute = minute.new_from_index(0, max_len);
     }
-    let minute = minute.u32()?;
+    let minute = minute.i8()?;
 
-    let mut second = second.cast(&DataType::UInt32)?;
+    let mut second = second.cast(&DataType::Int8)?;
     if second.len() < max_len {
         second = second.new_from_index(0, max_len);
     }
-    let second = second.u32()?;
+    let second = second.i8()?;
 
-    let mut microsecond = microsecond.cast(&DataType::UInt32)?;
+    let mut microsecond = microsecond.cast(&DataType::Int32)?;
     if microsecond.len() < max_len {
         microsecond = microsecond.new_from_index(0, max_len);
     }
-    let microsecond = microsecond.u32()?;
+    let microsecond = microsecond.i32()?;
     let mut _ambiguous = ambiguous.cast(&DataType::String)?;
     if _ambiguous.len() < max_len {
         _ambiguous = _ambiguous.new_from_index(0, max_len);
     }
-    let _ambiguous = _ambiguous.str()?;
-
-    let ca: Int64Chunked = year
-        .into_iter()
-        .zip(month)
-        .zip(day)
-        .zip(hour)
-        .zip(minute)
-        .zip(second)
-        .zip(microsecond)
-        .map(|((((((y, m), d), h), mnt), s), us)| {
-            if let (Some(y), Some(m), Some(d), Some(h), Some(mnt), Some(s), Some(us)) =
-                (y, m, d, h, mnt, s, us)
-            {
-                NaiveDate::from_ymd_opt(y, m, d)
-                    .and_then(|nd| nd.and_hms_micro_opt(h, mnt, s, us))
-                    .map(|ndt| match time_unit {
-                        TimeUnit::Milliseconds => ndt.and_utc().timestamp_millis(),
-                        TimeUnit::Microseconds => ndt.and_utc().timestamp_micros(),
-                        TimeUnit::Nanoseconds => ndt.and_utc().timestamp_nanos_opt().unwrap(),
-                    })
-            } else {
-                None
-            }
-        })
-        .collect_trusted();
-
-    let ca = match time_zone {
-        #[cfg(feature = "timezones")]
-        Some(_) => {
-            let mut ca = ca.into_datetime(*time_unit, None);
-            ca = replace_time_zone(&ca, time_zone, _ambiguous, NonExistent::Raise)?;
-            ca
-        },
-        _ => {
-            polars_ensure!(
-                time_zone.is_none(),
-                ComputeError: "cannot make use of the `time_zone` argument without the 'timezones' feature enabled."
-            );
-            ca.into_datetime(*time_unit, None)
-        },
-    };
```

Comment:
I moved this code into datetime_series_from_parts in polars_time/src/series/mod.rs for re-use.